File: rpg-backend/app/schemas/expedition.py
from pydantic import BaseModel


class ExpeditionTemplateOut(BaseModel):
    id: int
    name: str
    description: str | None
    image_path: str | None
    duration_seconds: int
    required_hero_level: int
    reward_xp: int
    reward_coins: int
    is_active: bool
    is_available_to_user: bool


class ExpeditionSummaryOut(BaseModel):
    id: int
    name: str


class UserExpeditionOut(BaseModel):
    id: int
    expedition: ExpeditionSummaryOut
    status: str  # "running" | "completed" | "claimed" — see ExpeditionStatus's docstring
    started_at: str
    completed_at: str
    claimed_at: str | None
    reward_xp: int
    reward_coins: int
    hero_level: int
    hero_xp: int
    balance: int

File: rpg-frontend/src/types/expedition.ts
export interface ExpeditionTemplateOut {
  id: number;
  name: string;
  description: string | null;
  image_path: string | null;
  duration_seconds: number;
  required_hero_level: number;
  reward_xp: number;
  reward_coins: number;
  is_active: boolean;
  is_available_to_user: boolean;
}

export interface ExpeditionSummaryOut {
  id: number;
  name: string;
}

export interface UserExpeditionOut {
  id: number;
  expedition: ExpeditionSummaryOut;
  status: "running" | "completed" | "claimed";
  started_at: string;
  completed_at: string;
  claimed_at: string | null;
  reward_xp: number;
  reward_coins: number;
  hero_level: number;
  hero_xp: number;
  balance: number;
}
